source/LocationFilter.cpp

```diff
@@ -71,6 +89,53 @@ namespace {
 		return false;
 	}
 
+	shared_ptr<DistanceMap> FlagshipMap(const PlayerInfo &player)
+	{
+		static mutex distanceMutex;
+		lock_guard<mutex> lock(distanceMutex);
+
+		// Storing a weak pointer ensures that when the entire tree of LocationFilter
+		// methods exits, the DistanceMap is destructed and will be replaced upon the
+		// next call to the same LocationFilter tree.
+		static weak_ptr<DistanceMap> map;
+
+		try {
+			return shared_ptr<DistanceMap>(map);
+		}
+		catch(const bad_weak_ptr &bad)
```

Comment:
Can this not be avoided by an `expired()` check?

Reply:
Using expired() introduces a race condition. It's a bad habit to get into.